collect-intent.rst

```diff
@@ -0,0 +1,181 @@
+  intent.setData(org.odk.collect.android.provider.FormsProviderAPI.FormsColumns.CONTENT_URI);
+ 
+.. code-block:: java
+ 
+  static final int PICK_FORM_REQUEST = 1;  // The request code
+  startActivityForResult(intent, PICK_FORM_REQUEST);
+
+.. _get-result:
+
+Getting result
+''''''''''''''''
+ 
+To get the result, override ``onActivityResultMethod`` in the followig way:
+
+.. code-block:: java
+
+  @Override
+  protected void onActivityResult(int requestCode, int resultCode, Intent formUri) {
+     // Check which request we're responding to
+    if (requestCode == PICK_FORM_REQUEST) {
+        // Make sure the request was successful
+	    if (resultCode == RESULT_OK) {
+ 	      // The user picked a contact.
+ 	      // The Intent's data Uri identifies which form was selected.
+ 	      // Do something with the form here
+ 	    }
+	}	
+  }
+ 
+
+Similarly for an instance, change the URI to that of the instance:
+ 
+.. code-block:: java
+ 
+  intent.setData(org.odk.collect.android.provider.InstanceProviderAPI.InstanceColumns.CONTENT_URI);
+
+.. _use-form-uri:
+
+Use form's URI to edit/view form
+""""""""""""""""""""""""""""""""""
+ 
+The formURI in the ``onActivityResult()`` method, allows us to view/edit the particular form by:
+ 
+.. code-block:: java
+ 
+  Intent intent = new Intent(android.intent.action.EDIT);
+  intent.setData(formUri);
+ 
+If we want to view the form, the action can be changed to:
+ 
+.. code-block:: java
+ 
+  Intent intent = new Intent(android.intent.action.VIEW);
+ 
+Similar things can be done for an Instance.
+
+.. _launch-apps:
+
+Launch External Apps from ODK Collect
+---------------------------------------
+
+ODK Collect can launch 3rd party apps to populate string, integer or numeric fields. Beginning with ODK Collect 1.4.3, an external app can populate a group of fields. Also beginning with ODK Collect 1.4.3, any number of additional values, beyond the current value(s) of the field(s) being updated, can be passed to the 3rd party app.
+
+- A text/decimal/integer field with an **ex:intentString** appearance can specify extra parameters that are passed to the external app, in addition to the ``value`` parameter that holds the current value for that field. The names of the parameters are user defined and there are no reserved names. 
+
+.. code-block:: xml
+
+  <input appearance="ex:org.myapp.COLLECT(started= /externaltest/starttime ,
+                                          constant='----', randomNumber=random())" 
+           ref="/externaltest/textField" >
+      <label>Click launch to see an external-fetched string</label>
+  </input>
+
+Any number of extra parameters can be specified. The parameter values can be four different things:
+
+  - An xpath expression to an other field.
+  - A string literal defined in single quotes.
+  - A raw number (integer or decimal)
+  - Any JavaRosa function.
+
+
+- A ``field-list`` group can also have an ``intent`` attribute.
+
+.. code-block:: xml
```

Comment:
As above: If possible, it would be good to show how to accomplish this in an XSLForm.